### PR TITLE
Install Smart Contract

### PR DESCRIPTION
**Pull request by caroline-church**
Install a smart contract on a peer
Add integration test
Update integration test to use master version of fabric

contributes to IBM-Blockchain/blockchain-vscode-extension#34

Signed-off-by: Caroline Church <caroline.church@uk.ibm.com>

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -1,400 +1,425 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/IBM-Blockchain/blockchain-vscode-extension"
+  },
+  "categories": [
+    "Programming Languages",
+    "Other"
+  ],
+  "activationEvents": [
```
Comment (sstone1) [inline client/package.json]: y u no leave as `*`

Comment (sstone1) [reply client/package.json]: also y u reformat

Comment (caroline-church) [reply client/package.json]: oops

File: client/src/commands/UserInputUtil.ts [hunk 2]
```diff
@@ -145,4 +145,14 @@ export class UserInputUtil {
 
         return vscode.window.showQuickPick(quickPickItems, quickPickOptions);
     }
+
+    public static showSmartContractLanguagesQuickPick(prompt: string, languages: Array<string>): Thenable<string | undefined> {
+        const choseSmartContractLanguageQuickPickOptions = {
```
Comment (sstone1) [inline client/src/commands/UserInputUtil.ts]: y u no type 😉

File: client/src/commands/createSmartContractProjectCommand.ts [hunk 3]
```diff
@@ -251,6 +248,7 @@ async function getSmartContractLanguageOptions(): Promise<string[]> {
 
 async function openNewProject(openMethod: string, uri: vscode.Uri): Promise<void> {
     if (openMethod === UserInputUtil.ADD_TO_WORKSPACE) {
+        console.log('OPEN WORKSPACE');
```
Comment (sstone1) [inline client/src/commands/createSmartContractProjectCommand.ts]: really?

File: client/src/commands/installCommand.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    if (!peerTreeItem) {
+        if (!FabricConnectionManager.instance().getConnection()) {
+            await vscode.commands.executeCommand('blockchainExplorer.connectEntry');
+            if (!FabricConnectionManager.instance().getConnection()) {
+                // either the user cancelled or ther was an error so don't carry on
+                return;
+            }
+        }
+
+        const chosenPeerName = await UserInputUtil.showPeerQuickPickBox('Choose a peer to install the chaincode on');
```
Comment (sstone1) [inline client/src/commands/installCommand.ts]: y u no type? 😉

File: client/src/commands/installCommand.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        peerName = peerTreeItem.peerName;
+    }
+
+    try {
+        const chosenPackage: IBlockchainQuickPickItem<PackageRegistryEntry> = await UserInputUtil.showSmartContractPackagesQuickPickBox('Choose which package to install on the peer', false) as IBlockchainQuickPickItem<PackageRegistryEntry>;
+        if (!chosenPackage) {
+            return;
+        }
+
+        const fabricClientConnection = FabricConnectionManager.instance().getConnection();
```
Comment (sstone1) [inline client/src/commands/installCommand.ts]: y u no type? 😉

**Pull request merged**